core/src/main/java/org/apache/accumulo/core/dataImpl/KeyExtent.java

```diff
@@ -457,6 +457,12 @@ private void decodeMetadataRow(Text flattenedExtent) {
     return ke.getTableId().canonical().getBytes(UTF_8);
   }
 
+  public static TableId tableIdOfMetadataRow(Text row) {
+    KeyExtent ke = new KeyExtent();
+    ke.decodeMetadataRow(row);
+    return ke.getTableId();
```

Comment:
Could replace this with

```java
 return new KeyExtent(row, (Text) null).getTableId();
```

Reply:
It looks like there's already a `tableOfMetadataRow` method. It returns a tableId as a `byte[]`... but the only two places where it is used converts it immediately back to a `new String()`.

Rather than create this new method, I would just change the old one to return the `TableId` object instead, and make its previous callers just call `tableId.canonical()` instead of using `new String()`.

Reply:
Actually, both the existing two places eventually turn it back into a `TableId` instance anyway... which is all a waste.

Reply:
I noticed that as well but didn't want these changes to get too out of scope.  I was looking at KeyExtent yesterday and think there could be some other cleanup as well.  I will open that as a follow on.

Reply:
could be a good first issue if what needs to be done is very specific.

Reply:
I'll just knock it out, so it makes this change simpler.